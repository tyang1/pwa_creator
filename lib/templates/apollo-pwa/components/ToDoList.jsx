import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardHeader from "@material-ui/core/CardHeader";
import ItemList from "../containers/List.jsx";

// import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
  },
}));

export default function ToDoList(props) {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardHeader title="To Do List" subheader="September 14, 2016" />
      <CardContent>
        <ItemList />
      </CardContent>
    </Card>
  );
}
