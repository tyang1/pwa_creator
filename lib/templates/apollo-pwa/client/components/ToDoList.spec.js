import React from "react";
import renderer from "react-test-renderer";
import ToDoList from "./ToDoList.jsx";
import ItemList from "../containers/List.jsx";
import { mount } from "enzyme";

describe("ToDoList", () => {
  it("should render componet", () => {
    const component = renderer.create(<ToDoList data={null} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render ItemList", () => {
    const wrapper = mount(
      <ToDoList data={{ todos: [{ name: "study more", completed: false }] }} />
    );
    expect(wrapper.find(ItemList).length).toEqual(1);
  });
  it("should pass data props to ItemList", () => {
    const wrapper = mount(
      <ToDoList data={{ todos: [{ name: "study more", completed: false }] }} />
    );
    expect(wrapper.find(ItemList).prop("data").todos).toHaveLength(1);
  });
});
