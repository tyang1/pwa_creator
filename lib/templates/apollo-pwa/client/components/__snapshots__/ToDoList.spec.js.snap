// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList should render componet 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        To Do List
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        September 14, 2016
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <ul
      className="MuiList-root MuiList-padding"
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="standard-name"
          id="standard-name-label"
        >
          Add To Do
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="standard-name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Save To Do
        </span>
      </button>
    </ul>
  </div>
</div>
`;
